### PR TITLE
docs: v2 migration guides, docs, changesets

### PR DESCRIPTION
**Pull request by DanielSinclair**

### CODE REVIEW

File: site/data/en-US/docs/migration-guide.mdx [hunk 1]
```diff
@@ -6,6 +6,316 @@ title: Migration Guide
...
++      })(config),
++      ...walletDetails,
++    }));
++  },
+});
+```
+
+**5. Custom chains**
+
+We no longer support custom chains. You no longer can pass `chains` into `<RainbowKitProvider>`
```
Comment (Songkeys) [inline site/data/en-US/docs/migration-guide.mdx:316]: I think this should be stated more clearly - we can still provide custom chains via wagmi config while there is no need to pass into here.

Also I found that custom`iconUrl` is no longer supported?

Comment (oxSaturn) [reply site/data/en-US/docs/migration-guide.mdx:316]: Yeah, just curious, without `iconUrl`, how will we include icons for chains that's are not built-in in rainbowkit?

CleanShot-AFdaZkqR@2x

**Review by ITZSHOAIB — APPROVED** [review]

File: site/data/en-US/docs/migration-guide.mdx (hunk 1 above)
Comment (KosmosKey) [reply site/data/en-US/docs/migration-guide.mdx:316]: Good catch! We'll introduce a custom api `getChains` to provide custom background color and icon url

File: site/data/en-US/docs/migration-guide.mdx [hunk 2]
```diff
@@ -6,6 +6,373 @@ title: Migration Guide
...
+})
+
++ const queryClient = new QueryClient()
+
+const App = () => {
+  return (
+-    <WagmiConfig config={config}>
++    <WagmiProvider config={config}>
++      <QueryClientProvider client={queryClient}>
+        <RainbowKitProvider chains={chains}>
```
Comment (Songkeys) [inline site/data/en-US/docs/migration-guide.mdx:68]: There is no `chains` arg now, isn't it?

**Review by songkeys — APPROVED** [review]

**Pull request merged**